CatalogService.Infrastructure/MsSql/Categories/CategoryRepository.cs

```diff
@@ -0,0 +1,142 @@
+﻿using CatalogService.Infrastructure.Dto;
+using CatalogService.Infrastructure.MsSql.DatabaseModels;
+using Microsoft.EntityFrameworkCore;
+
+namespace CatalogService.Infrastructure.MsSql.Categories;
+
+public interface ICategoryRepository
```

Comment:
In general it would be nice to have interfaces in a separate files (and if you are having dependency inversion even in the separate projects)

Reply:
Extracted interfaces